swift/ql/lib/codeql/swift/frameworks/StandardLibrary/CustomUrlSchemes.qll

```diff
@@ -56,7 +56,9 @@ private class ApplicationWithLaunchOptionsFunc extends Function {
 
 private class LaunchOptionsUrlVarDecl extends VarDecl {
   LaunchOptionsUrlVarDecl() {
-    this.getEnclosingDecl().asNominalTypeDecl().getFullName() = "UIApplication.LaunchOptionsKey" and
+    // ideally this would be the more accurate, but currently less robust:
+    // this.getEnclosingDecl().asNominalTypeDecl().getFullName() = "UIApplication.LaunchOptionsKey" and
+    this.getType().getName() = "UIApplication.LaunchOptionsKey" and
```

Comment:
Should this be `getFullName`?

Reply:
At present `Type.getName()` appears to return the full name for the `Type`, at least in the case of simply nested nominal types - this is probably not ideal.  There is also `NominalType.getFullName()` that does the same.  For `NominalTypeDecl` the two exist as well and behave as you'd expect.

I'll look into this some more and hopefully fix it tomorrow.

Reply:
Yeah, `Type.getName` gives all sorts of interesting strings, such as:
```
Builtin.IntLiteral
Decodable & Encodable
Realm.UpdatePolicy
```
Some of these might be the most reasonable thing we can do, but the last one probably isn't.  I'll write up an issue about it.

In the last example only the `NominalTypeDecl.getName` is `UpdatePolicy`.  `Type.getName` gives the full name along with `NominalType.getFullName` and `NominalTypeDecl.getFullName`.  I guess the thing to do is to use `NominalType.getFullName` here so that this doesn't need to be changed if `getName` is fixed.

Reply:
Fixed to `(NominalType).getFullName()`.